lighthouse-cli/test/cli/artifacts-test.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* eslint-env jest */
+
+const testServer = require('../fixtures/static-server.js').server;
+const run = require('../../run.js');
+const getFlags = require('../../cli-flags').getFlags;
+
+/** @type {Array<keyof LH.Artifacts>} */
+const ARTIFACTS_TO_ASSERT = [
+  'AnchorElements',
+  'ImageElements',
+  'LinkElements',
+  'MetaElements',
+  'ScriptElements',
+];
+
+/** @type {Array<[(path: string[], value: any) => boolean, (value: any) => any]>} */
+const cleaningFunctions = [
+  [path => path[0] === 'requestId', () => '<requestId>'],
+  [path => path[0] === 'content', value => value && value.slice(0, 100)],
+  [(_, value) => /localhost:\d+/.test(value), value => value.replace(/localhost:\d+/, 'localhost')],
+  [(_, value) => /^blob:/.test(value), () => '<blob url>'],
+];
```

Comment:
can you add a comment (or comments) for each of these

Reply:
sure thing if we want to proceed!